src/components/Toast/__stories__/toast.stories.mdx

```diff
@@ -0,0 +1,157 @@
+import { ArgsTable, Story, Canvas, Meta } from "@storybook/addon-docs";
+import { useState } from "react";
+import Toast from "../Toast";
+import { createComponentTemplate } from "../../../storybook/functions/create-component-story";
+import Button from "../../Button/Button.jsx";
+import { Link } from "../../../storybook/components";
+import { TOOLTIP, ATTENTION_BOX, ALERT_BANNER } from "../../../storybook/components/related-components/component-description-map";
+import "./toast.stories.scss";
+
+<Meta
+  title="Components/Toast"
+  component={ Toast }
+/>
+
+<!--- Component template -->
+
+export const toastTemplate = createComponentTemplate(Toast);
+
+<!--- Component documentation -->
+
+# Toast
+- [Overview](#overview)
+- [Props](#props)
+- [Usage](#usage)
+- [Variants](#variants)
+- [Do’s and don’ts](#dos-and-donts)
+- [Use cases and examples](#use-cases-and-examples)
+- [Related components](#related-components)
+- [Feedback](#feedback)
+
+## Overview
+A toast notification is a message object that presents timely information, including confirmation of actions, alerts, and errors.
+
+<Canvas>
+  <Story name="Overview"
+         args={ {children: "General message toast", open: true, className: "monday-storybook-toast_wrapper", actions: [{ type: Toast.actionTypes.BUTTON, content: "Button" }]} }>
+    { toastTemplate.bind({}) }
+  </Story>
+</Canvas>
+
+## Props
+<ArgsTable of={ Toast } />
+
+## Usage
+<UsageGuidelines guidelines={[
+  "Use toast notifications immediately after a specific event such as a user action that does not relate to an object on the page. Toast used as a feedback loop to a user’s action.",
+  "Toasts should appear one at a time, and only disappear when no longer required.",
+  "Always be clear, concise and, where possible, give follow up actions to allow the user to become more informed or resolve the issue.",
+  "Always provide an action button or option to undo.",
+  "Toast should overlay permanent UI elements without blocking important actions.",
+]}/>
+
+<Tip title="Check yourself">Need to inform the user about a system’s action? Use an <Link href="/?path=/docs/components-alertbanner--overview" size={Link.sizes.SMALL} withoutSpacing>Alert banner</Link> instead.</Tip>
+
+## Variants
+### Default with button
+<Canvas>
+  <Story name="Default with button">
+    <Toast open className="monday-storybook-toast_wrapper" actions={[{ type: Toast.actionTypes.BUTTON, content: "Button" }]}>General message toast</Toast>
```

Comment:
please create the action array by using useMemo.
for doing this you will need to write the child of the story as anonymous function. you can see a reference for similar implementation in the button story mdx under the story: "success state."